lib/journaled/transaction_ext.rb

```diff
@@ -0,0 +1,82 @@
+require 'active_record/connection_adapters/abstract/transaction'
+
+module Journaled
+  module TransactionExt
+    def before_commit_records
+      @_journaled_committing_records = true
+      super
+    end
+
+    def _journaled_transaction_handler
+      if @_journaled_transaction_handler&.active?
```

Comment:
Do we still need to track active-ness now that it's tied to the transaction? I don't think there's a way to enqueue an event on a transaction after it's been committed or rolledback (since it would no longer be the current transaction, right?).

If we do keep it, I think we should probably error if it's accessed after we've committed or rolledback to prevent events from being added when they won't be sent.

Reply:
Good point!